### PR TITLE
feature(C04P09):add solution

### PR DESCRIPTION
**Pull request by ryoyama0508**
Add a backtracking solution to `chapter_04/p09_bst_sequences.py`.

### CODE REVIEW

File: chapter_04/p09_bst_sequences.py [hunk 1]
```diff
@@ -39,6 +39,34 @@ def weave(first, second, prefix, results):
...
+ret_backtracking = []  # global variable for backtracking
+
+
+def find_bst_sequences_backtracking(bst):
+    if not bst.root:
+        return []
+    return helper2(bst.root)
+
+
+def helper2(node):
```
Comment (brycedrennan) [inline chapter_04/p09_bst_sequences.py:51]: this function doesn't seem to need to be split from the main fucntion

File: chapter_04/p09_bst_sequences.py [hunk 2]
```diff
@@ -39,6 +39,34 @@ def weave(first, second, prefix, results):
     return results
 
 
+ret_backtracking = []  # global variable for backtracking
```
Comment (brycedrennan) [inline chapter_04/p09_bst_sequences.py:42]: if you nest the function definitions then you can keep this out of the global scope

**Review by brycedrennan — APPROVED** [review]
Thanks! I've merged your work, with some minor changes, here: 
https://github.com/careercup/CtCI-6th-Edition-Python/commit/9f37fcbee4ded7e62c655fb3c1ca724470a931e0

**Comment by ryoyama0508** [comment]
Thanks! Looks cleaner than my code. haha :)

**Pull request closed without merging**